OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -132,14 +139,38 @@ void IBotTick.BotTick(IBot bot)
 
 		bool ShouldBuildMCV()
 		{
-			// Only build MCV if we don't already have one in the field.
-			var allowedToBuildMCV = AIUtils.CountActorByCommonName(Info.McvTypes, player) == 0;
-			if (!allowedToBuildMCV)
+			if (world.WorldTick < newMcvInterval)
+				return false;
+
+			var enoughFactories = AIUtils.CountBuildingByCommonName(Info.McvFactoryTypes, player) > 0;
+			if (!enoughFactories)
+				return false;
+
+			var mcvCount = AIUtils.CountActorByCommonName(Info.McvTypes, player);
+			if (mcvCount > 0)
+				return false;
+
+			var conyardCount = AIUtils.CountBuildingByCommonName(Info.ConstructionYardTypes, player);
+			if (conyardCount >= Info.MinimumConstructionYardCount)
 				return false;
 
-			// Build MCV if we don't have the desired number of construction yards, unless we have no factory (can't build it).
-			return AIUtils.CountBuildingByCommonName(Info.ConstructionYardTypes, player) < Info.MinimumConstructionYardCount &&
-				AIUtils.CountBuildingByCommonName(Info.McvFactoryTypes, player) > 0;
+			if (conyardCount > 0 && world.WorldTick < buildAdditionalMcvDelay)
+				return false;
+
+			// Only build MCV if there isn't already one being built.
+			var queues = player.World.ActorsWithTrait<ProductionQueue>()
+				.Where(a => a.Actor.Owner == player && Info.McvFactoryTypes.Contains(a.Actor.Info.Name) && a.Trait.Enabled)
+				.Select(a => a.Trait);
+
+			foreach (var factory in queues)
+				if (factory.AllQueued().Any(z => Info.McvTypes.Contains(z.Item)))
+				{
+					newMcvInterval = world.WorldTick + 2000;
```

Comment:
Shouldn't this be unhardcoded too? Also why are we just resetting this timer together with `buildAdditionalMcvDelay`?

Reply:
It's a performance tick to ensure that the AI doesn't try to request another MCV within the current WorldTick + 2000. The other parameter buildAdditionalMcvDelay is reset to WorldTick + configurable BuildAdditionalMcvDelay. It allows you to slow down the rate at which the AI is ultimately able to request new construction yards.

Reply:
Yes, but why not 
```suggestion
					newMcvInterval = world.WorldTick + Info.BuildMcvDelay;
```